File: MasterCatalog.Api/Services/UnitOfMeasureService.cs
using MasterCatalog.Dal.Contracts;
using MasterCatalog.Domain.Models;

namespace MasterCatalog.Api.Services
{
    public class UnitOfMeasureService : IUnitOfMeasureService
    {
        private readonly IUnitOfMeasureRepository _unitOfMeasureRepository;

        public UnitOfMeasureService(IUnitOfMeasureRepository unitOfMeasureRepository)
        {
            _unitOfMeasureRepository = unitOfMeasureRepository;
        }

        public List<UnitOfMeasure> GetUnitsOfMeasure()
        {
            return _unitOfMeasureRepository.GetAll();
        }

        public Dictionary<int, UnitOfMeasure> GetUnitsOfMeasureGroupedByUnitOfMeasureID()
        {
            return GetUnitsOfMeasure().ToDictionary(x => x.UnitOfMeasureID.Value, y => y);
        }

        public UnitOfMeasure InsertUnitOfMeasure(UnitOfMeasure unitOfMeasure)
        {
            _unitOfMeasureRepository.Insert(unitOfMeasure);
            return unitOfMeasure;
        }

        public UnitOfMeasure UpdateUnitOfMeasure(UnitOfMeasure unitOfMeasure)
        {
            _unitOfMeasureRepository.Update(unitOfMeasure);
            return unitOfMeasure;
        }

        public void DeleteUnitOfMeasure(int unitOfMeasureID)
        {
            _unitOfMeasureRepository.Delete(unitOfMeasureID);
        }
    }
}

File: MasterCatalog.Web/wwwroot/js/services/unitOfMeasureService.js
import { ApiUtility } from '../utilities/apiHelper.js';
import { UnitOfMeasure } from '../models/unitOfMeasure.js';

const _api = new ApiUtility('unitofmeasure');

export class UnitOfMeasureService {

    async insert(obj) {
        const response = await _api.insert(obj);
        return this.mapToModel(response.data);
    }

    async update(obj) {
        const response = await _api.update(obj);
        return this.mapToModel(response.data);
    }

    async delete(id) {
        return _api.delete(id);
    }

    async get() {
        const response = await _api.get();
        const uoms = response.data.map((obj) => {
            return this.mapToModel(obj);
        });

        return uoms;
    }

    mapToModel(obj) {
        const uom = new UnitOfMeasure(
            obj.unitOfMeasureID,
            obj.unitOfMeasureName,
            obj.unitOfMeasureDescription
        );

        return uom;
    }
}